From e8b18b9cc2e92d7f9e4bdccdea40c224e742fb1f Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <i.gnatenko.brain@gmail.com>
Date: Tue, 25 Jul 2017 21:04:58 +0200
Subject: [PATCH] macros: enable debugsource and debuginfo-subpkgs by default

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>
---
 macros.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/macros.in b/macros.in
index 851822891..f71fcc103 100644
--- a/macros.in
+++ b/macros.in
@@ -549,10 +549,10 @@ package or when debugging this package.\
 %_unique_debug_srcs	1
 
 # Whether rpm should put debug source files into its own subpackage
-#%_debugsource_packages	1
+%_debugsource_packages	1
 
 # Whether rpm should create extra debuginfo packages for each subpackage
-#%_debuginfo_subpackages 1
+%_debuginfo_subpackages 1
 
 #
 # Use internal dependency generator rather than external helpers?
-- 
2.13.3

