From cbbbf06a0ba3aaced56edf5f632c1b0da6da0c6b Mon Sep 17 00:00:00 2001
Message-Id: <cbbbf06a0ba3aaced56edf5f632c1b0da6da0c6b.1586433208.git.pmatilai@redhat.com>
From: Panu Matilainen <pmatilai@redhat.com>
Date: Thu, 9 Apr 2020 14:49:30 +0300
Subject: [PATCH] Fix regression causing all ELF files classified as OCaml

Commit a6fe37c39b39acbcbd014dd1e6d5653ff84254a1 causes OCaml generators
to execute on all ELF files due to missing "magic_and_path" flag.

Fixes: #1173
---
 fileattrs/ocaml.attr | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fileattrs/ocaml.attr b/fileattrs/ocaml.attr
index ad698b5fa..4bd65c17a 100644
--- a/fileattrs/ocaml.attr
+++ b/fileattrs/ocaml.attr
@@ -2,3 +2,4 @@
 %__ocaml_requires	%{_rpmconfigdir}/ocamldeps.sh --requires
 %__ocaml_magic		^(ELF|Objective caml|OCaml) .*$
 %__ocaml_path		.(cma|cmi|cmo|cmx|cmxa|cmxs)$
+%__ocaml_flags		magic_and_path
-- 
2.25.2

