diff -up rpm-4.17.1/macros.in.orig rpm-4.17.1/macros.in
--- rpm-4.17.1/macros.in.orig	2022-08-02 14:19:36.496729048 +0200
+++ rpm-4.17.1/macros.in	2022-08-02 14:19:56.270615191 +0200
@@ -1166,10 +1166,7 @@ package or when debugging this package.\
 %{__git} config gc.auto 0\
 %{__git} add --force .\
 %{__git} commit %{-q} --allow-empty -a\\\
-	--author "%{__scm_author}" -m "%{NAME}-%{VERSION} base"\
-%{__git} branch rpm-build \
-%{__git} checkout rpm-build \
-%{__git} branch --set-upstream-to=master
+	--author "%{__scm_author}" -m "%{NAME}-%{VERSION} base"
 
 %__scm_apply_git(qp:m:)\
 %{__git} apply --index --reject %{-p:-p%{-p*}} -\
