diff -up rpm-4.9.1.3/build/files.c.doc-quotes rpm-4.9.1.3/build/files.c
--- rpm-4.9.1.3/build/files.c.doc-quotes	2013-03-27 08:04:35.117556856 +0200
+++ rpm-4.9.1.3/build/files.c	2013-03-27 08:04:35.143556768 +0200
@@ -892,7 +892,9 @@ static rpmRC parseForSimple(rpmSpec spec
 
 	if (*s != '/') {
 	    if (fl->currentFlags & RPMFILE_DOC) {
-		rstrscat(&specialDocBuf, " ", s, NULL);
+		char * fn = rpmEscapeSpaces(s);
+		rstrscat(&specialDocBuf, " ", fn, NULL);
+		free(fn);
 	    } else
 	    if (fl->currentFlags & RPMFILE_PUBKEY)
 	    {
